org.compiere.db.oracle.provider/src/org/adempiere/db/oracle/config/ConfigOracle.java

```diff
@@ -320,22 +320,22 @@ public String test(IDBConfigMonitor monitor, ConfigurationData data)
 		}
 		if (monitor != null)
 			monitor.update(new DBConfigStatus(DBConfigStatus.DATABASE_SERVER, "ErrorDatabaseServer",
-				pass, true, error));
+				pass, false, error));
 		if (log.isLoggable(Level.INFO)) log.info("OK: Database Server = " + databaseServer);
-		data.setProperty(ConfigurationData.ADEMPIERE_DB_SERVER, databaseServer!=null ? databaseServer.getHostName() : null);
+		data.setProperty(ConfigurationData.ADEMPIERE_DB_SERVER, databaseServer!=null ? databaseServer.getHostName() : "");
 		//store as lower case for better script level backward compatibility
 		data.setProperty(ConfigurationData.ADEMPIERE_DB_TYPE, data.getDatabaseType());
 		data.setProperty(ConfigurationData.ADEMPIERE_DB_PATH, data.getDatabaseType().toLowerCase());
 
 		//	Database Port
 		int databasePort = data.getDatabasePort();
-		pass = data.testPort (databaseServer, databasePort, true);
+		pass = pass ? data.testPort (databaseServer, databasePort, true) : false;
 		error = "DB Server Port = " + databasePort;
 		if (monitor != null)
 			monitor.update(new DBConfigStatus(DBConfigStatus.DATABASE_SERVER, "ErrorDatabasePort",
-				pass, true, error));
-		if (!pass)
-			return error;
+				pass, false, error));
+		// if (!pass)
+		// 	return error;
```

Comment:
same as above, invalid or missing DB server is a critical error if we are using the traditional method to connect to Oracle DB.